pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -41,6 +40,29 @@ func createPipelineSpec(maven bool, namespace string, commitTime int64) *pipelin
 		build = gradleBuild
 	}
 	zero := int64(0)
+	deployArgs := []string{
```

Comment:
another education question ... why do we need this config in the pipelinespec no?  was this an overisght when I moved off of configmaps?

if not, I'm missing how the kcp results pivot necessitates this.  As with some of my other question, I'm not saying this is wrong, I just need help understanding.  Thanks.

Reply:
Basically now that we want to stop the cache using the Java kube API as much as possible there is no need to involve it in the deployment process at all, so it has just been moved to a discrete step on task. The results of the deployment need to be expressed as results on the PipelineRun, so there were really two choices:

- Post deployment to cache, cache deploys, returns results as JSON, write JSON status update to results
- Deploy directly from pipeline

The originally moved to cache vs sidecar as it would let us avoid mounting the service token in the PR, however that is no longer an option as we need it for the results.

This approach also means that any problems are logged in the PR instead of just getting a 500 response, and any additional memory needed by the deploy step is only used briefly during the deploy process.

The second approach is much simpler ()

Reply:
OK I think I missed where we moved this stuff off of java code 

Makes sense / I'll see if I can connect those dots when I revisit the tomorrow after processing the responses here